server/order/controller/product_controller.js

```diff
@@ -5,8 +5,17 @@
 const Product = require('../model/product');
 
 exports.addProduct = function (productData) {
-  const product = new Product(JSON.parse(productData));
-  product.save();
+  return new Promise((resolve, reject) => {
+    Product.findOneAndUpdate({ name: productData.name }
```

Comment:
IMHO, the name can not be unique key. For example, there are two types of Americano, which has a different coffee bean. (This example is a real world example in coffee store in front of my house.)

But as-is okay for now. (in this branch) I hope we should fix it before cherry-picking master.